src/Kiota.Builder/CodeRenderer.cs

```diff
@@ -42,9 +42,14 @@ public async Task RenderCodeNamespaceToFilePerClassAsync(LanguageWriter writer,
                     {
                         var namespaceNameLastSegment = codeNamespace.Name.Split('.').Last().ToLowerInvariant();
                         // if the module already has a class with the same name, it's going to be declared automatically
-                        if (_configuration.ShouldWriteBarrelsIfClassExists ||
-                            codeNamespace.FindChildByName<CodeClass>(namespaceNameLastSegment, false) == null)
+                        if (_configuration.ShouldWriteBarrelsIfClassExists && _configuration.setCodeRenderingCondition(codeNamespace))
+                         // TODO : Verify and plug the following condition in the language specific index rendering condition 
+                        //codeNamespace.FindChildByName<CodeClass>(namespaceNameLastSegment, false) == null)
```

Comment:
@baywet I need some help in understanding the commented condition. 
The condition was causing empty index files to be generated. 

Added logic to have custom index generation condition.

Reply:
Added a separate  TSCodeRenderer  file